gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptorHandler.java

```diff
@@ -571,13 +579,49 @@ private static File resolveProviderConfigurationReference(final String reference
             if (topologyFilename == null) {
                 topologyFilename = desc.getCluster();
             }
+
             topologyDescriptor = new File(destDirectory, topologyFilename + ".xml");
 
-            try (OutputStream outputStream = Files.newOutputStream(topologyDescriptor.toPath());
-                 OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
-                 BufferedWriter fw = new BufferedWriter(outputStreamWriter)) {
-              fw.write(sw.toString());
-              fw.flush();
+            // KNOX-2302
+            boolean persistGenerated = false;
+            // Determine if the topology already exists.
+            if (topologyDescriptor.exists()) {
+                // If it does, only overwrite it if it has changed (KNOX-2302)
+                // Compare the generated topology with the in-memory topology
+                Topology existing = null;
+                TopologyService topologyService = gwServices.getService(ServiceType.TOPOLOGY_SERVICE);
+                for (Topology t : topologyService.getTopologies()) {
+                    if (topologyFilename.equals(t.getName())) {
+                        existing = t;
+                        break;
+                    }
```

Comment:
Maybe introducing `topologyService.getTopology(String topologyName)`?

Reply:
I feel this is out of scope. It is not required for this PR.